eksupgrade/src/preflight_module.py

```diff
@@ -124,7 +124,8 @@ def get_cluster_version(
                 return
         cmk_key_check(errors, cluster_name, region, cluster_details, report, customer_report)
         security_group_check(errors, cluster_name, region, cluster_details, report, customer_report)
-        pod_security_policies(errors, cluster_name, region, report, customer_report)
+        if float(update_version) < 1.25:
```

Comment:
hmm, I'm not sure about this. If a cluster is on 1.24, we want to report on any PSPs in use before upgrading to 1.25

```
The PodSecurityPolicy (PSP) was deprecated in Kubernetes version 1.21 and removed in Kubernetes 1.25. PSPs are being replaced with [Pod Security Admission (PSA)](https://kubernetes.io/docs/concepts/security/pod-security-admission/), a built-in admission controller that implements the security controls outlined in the [Pod Security Standards (PSS) ](https://kubernetes.io/docs/concepts/security/pod-security-standards/). PSA and PSS have both reached beta feature states, and are enabled in Amazon EKS by default. To address PSP removal in 1.25, we recommend that you implement PSS in Amazon EKS. For more information, see [Implementing Pod Security Standards in Amazon EKS](http://aws.amazon.com/blogs/containers/implementing-pod-security-standards-in-amazon-eks/) on the AWS blog.
```

https://docs.aws.amazon.com/eks/latest/userguide/pod-security-policy.html

Reply:
whether or not it prevents you from upgrading is a topic for a different discussion - you can upgrade just fine with resources on the cluster that were created with deprecated or removed APIs, you just can't re-post those again after upgrade. 

however, the main point here is that if we are reporting on PSPs, then reporting before upgrading to 1.25 is the most critical juncture since 1.25 is when they are removed. The logic proposed here currently is silently hiding that reporting which can give users the impression that "all is good"

Reply:
This change wouldn't have worked either way with the K8s client api removing the method we are calling with the latest version in our dependency. All earlier upgrades would fail with the same reason.